missions/views/default_assigments.py

```diff
@@ -18,10 +18,11 @@ def get_success_url(self):
         return reverse('default_assignment_list', kwargs={'ship_id': ship_id})
 
     def form_valid(self, form):
```

Comment:
This method is called *after* the form has been validated. Imho using the fields attribute on Views is ok for very simple forms but anything that is more complex (more validation) should have a dedicated form_class.

Passing request context might be painful, but there is the `get_form_kwargs` hook.